diff --git a/node_modules/react-native-fbsdk/android/src/main/java/com/facebook/reactnative/androidsdk/FBSDKPackage.java b/node_modules/react-native-fbsdk/android/src/main/java/com/facebook/reactnative/androidsdk/FBSDKPackage.java
index af2c3a9..ab8e9d9 100644
--- a/node_modules/react-native-fbsdk/android/src/main/java/com/facebook/reactnative/androidsdk/FBSDKPackage.java
+++ b/node_modules/react-native-fbsdk/android/src/main/java/com/facebook/reactnative/androidsdk/FBSDKPackage.java
@@ -59,11 +59,6 @@ public class FBSDKPackage implements ReactPackage {
     }
 
     @Override
-    public List<Class<? extends JavaScriptModule>> createJSModules() {
-        return Collections.emptyList();
-    }
-
-    @Override
     public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
         return Arrays.<ViewManager>asList(
                 new FBLikeViewManager(),
diff --git a/node_modules/react-native-fbsdk/node_modules/.bin/react-native b/node_modules/react-native-fbsdk/node_modules/.bin/react-native
new file mode 120000
index 0000000..8158e87
--- /dev/null
+++ b/node_modules/react-native-fbsdk/node_modules/.bin/react-native
@@ -0,0 +1 @@
+../../../react-native/local-cli/wrong-react-native.js
\ No newline at end of file
